starlite/contrib/sqlalchemy/init_plugin/config/asyncio.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING, cast
+
+from sqlalchemy.ext.asyncio import AsyncConnection, AsyncEngine, AsyncSession, async_sessionmaker, create_async_engine
+
+from starlite.types import Empty
+from starlite.utils import (
+    delete_starlite_scope_state,
+    get_starlite_scope_state,
+)
+
+from .common import SESSION_SCOPE_KEY, SESSION_TERMINUS_ASGI_EVENTS, GenericSessionConfig, GenericSQLAlchemyConfig
+
+if TYPE_CHECKING:
+    from typing import Any, Callable
+
+    from sqlalchemy.orm import Session
+
+    from starlite.datastructures.state import State
+    from starlite.types import BeforeMessageSendHookHandler, EmptyType, Message, Scope
+
+__all__ = ("SQLAlchemyAsyncConfig", "AsyncSessionConfig")
+
+
+async def default_before_send_handler(message: Message, _: State, scope: Scope) -> None:
+    """Handle closing and cleaning up sessions before sending.
+
+    Args:
+        message: ASGI-``Message``
+        _: A ``State`` (not used)
+        scope: An ASGI-``Scope``
+
+    Returns:
+        None
+    """
+    session = cast("AsyncSession | None", get_starlite_scope_state(scope, SESSION_SCOPE_KEY))
+    if session and message["type"] in SESSION_TERMINUS_ASGI_EVENTS:
+        await session.close()
+        delete_starlite_scope_state(scope, SESSION_SCOPE_KEY)
+
+
+@dataclass
+class AsyncSessionConfig(GenericSessionConfig[AsyncConnection, AsyncEngine, AsyncSession]):
+    """SQLAlchemy async session config."""
+
+    sync_session_class: type[Session] | None | EmptyType = Empty
+
+
+@dataclass
+class SQLAlchemyAsyncConfig(GenericSQLAlchemyConfig[AsyncEngine, AsyncSession, async_sessionmaker]):
+    """Async SQLAlchemy Configuration."""
+
+    create_engine_callable: Callable[[str], AsyncEngine] = create_async_engine
+    """Callable that creates an :class:`AsyncEngine <sqlalchemy.ext.asyncio.AsyncEngine>` instance or instance of its
+    subclass.
+    """
+    session_config: AsyncSessionConfig = field(default_factory=AsyncSessionConfig)
+    """Configuration options for the ``sessionmaker``.
```

Comment:
```suggestion
    """Configuration options for the :class:`sessionmaker <sqlalchemy.orm.sessionmaker>`.
```
(Or at least I believe this is how it's linked)